docs/deployment-examples/custom-scripts/scripts-in-packages/index.md

```diff
@@ -19,21 +23,43 @@ Octopus will detect these scripts and invoke them at the appropriate time during
 2. Name each script to match the naming convention depending when you want the script to execute.
 3. Include these scripts at the root of your package. Octopus does not search subdirectories.
 
-:::hint
-Avoid duplicate scripts into your package, like `PreDeploy.sh` and `PreDeploy.ps1`, hoping Octopus will choose the right script on your behalf. Octopus will try to execute both of these scripts during the pre-deploy phase of the step since both bash and PowerShell are cross-platform runtimes. This might lead to surprising behavior.
-:::
-
 ## Running a script when a step fails
 
 You can create a file named `DeployFailed.<ext>`, which will be invoked if the step fails. Our blog post about this feature [describes how DeployFailed.<ext> works](https://octopus.com/blog/deployfailed).
 
+## How Octopus executes your scripts
+
+At each stage during the deployment, Octopus will look for a scripts matching the current stage, and execute the first matching script it finds ordered by a platform-specific priority (see [cross-platform support](#cross-platform-support)).
+
+1. Octopus extracts the package to new uniquely named directory. **This becomes the current working directory.**
```

Comment:
This is slightly duplicated with another page; but has the details pertinent to the script execution like the current working directory.